locations/spiders/tmobile_us.py

```diff
@@ -1,69 +1,27 @@
 
 
-class TmobileUSSpider(scrapy.Spider):
+class TmobileUSSpider(SitemapSpider, StructuredDataSpider):
     name = "tmobile_us"
     item_attributes = {
         "brand": "T-Mobile",
         "brand_wikidata": "Q3511885",
     }
+    sitemap_urls = ["https://www.t-mobile.com/stores/sitemap-business-main-pages.xml"]
+    sitemap_rules = [(r"/stores/[a-z]{2}/t-mobile-", "parse_sd")]
     allowed_domains = ["www.t-mobile.com"]
-    download_delay = 0.5
-
-    def start_requests(self) -> Iterator[Request]:
-        for lat, lon in point_locations(
-            "us_centroids_25mile_radius_state.csv",
-        ):
-            params = {
-                "INTNAV": "tNav:StoreLocator",
-                "lat": "{}".format(lat),
-                "lon": "{}".format(lon),
-                "showall": "false",
-            }
-
-            yield self.do_request(BASE_URL, params)
-
-    def do_request(self, url, params) -> Request:
-        return Request(url + urlencode(params), callback=self.parse, meta=params)
-
-    def parse(self, response: Response, **kwargs) -> Iterator[Feature] | Iterator[Request]:
-        data = response.json()
-
-        for poi in data.get("business_list", {}).get("object_list", []):
-            item = DictParser.parse(poi)
-            item["addr_full"] = " ".join(poi.get("address_text_lines", []))
-            item["state"] = poi.get("locale", {}).get("region", {}).get("state")
-            item["website"] = item["ref"] = urljoin("https://www.t-mobile.com/", poi["business_link"])
-            self.parse_hours(item, poi)
-            yield item
+    drop_attributes = {"facebook", "twitter"}
 
-        if next_page := data.get("business_list", {}).get("next_page_number"):
-            params = response.meta
-            params["page"] = next_page
-            yield self.do_request(BASE_URL, params)
+    def pre_process_data(self, ld_data: dict, **kwargs):
+        ld_data["openingHours"] = ld_data.pop("openingHoursSpecification", None)
 
-    def parse_hours(self, item: Feature, poi: dict):
-        hours = poi.get("all_opening_hours", {}).get("schemaHrs", [])
-        if not hours:
-            return
-        try:
-            oh = OpeningHours()
-            for hour in hours:
-                day, time = hour.split(" ")
-                time_open, time_close = time.split("-")
-                oh.add_range(day, time_open, time_close)
-            item["opening_hours"] = oh.as_opening_hours()
-        except Exception as e:
-            self.logger.warning(f"Failed to parse hours {hours}: {e}")
-            self.crawler.stats.inc_value(f"atp/{self.name}/hours/failed")
+    def post_process_item(self, item: Feature, response: Response, ld_data: dict, **kwargs):
+        item["addr_full"] = item.pop("street_address", None)
+        item["image"] = (
+            ld_data["image"].split("(webp)")[1].strip("/") if "(webp)" in ld_data["image"] else ld_data["image"]
+        )
+        yield item
```

Comment:
May I kindly ask to hardcode category please?

```suggestion
        item["image"] = (
            ld_data["image"].split("(webp)")[1].strip("/") if "(webp)" in ld_data["image"] else ld_data["image"]
        )
        apply_category(Categories.SHOP_MOBILE_PHONE, item)
        yield item
```

Reply:
Any specific reason to hardcode category here, when it's already available through `NSI`?